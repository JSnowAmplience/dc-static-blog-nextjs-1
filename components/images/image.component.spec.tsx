import Image from './image.component';
import AmplienceImage from '../../common/interfaces/image.interface';
import { MediaType } from '../../common/interfaces/media.interface';

import ShallowRenderer from 'react-test-renderer/shallow';

describe('Image', (): void => {
  let renderer;
  beforeEach(() => {
    renderer = ShallowRenderer.createRenderer();
  });
  it('should render a lazy image with width only', (): void => {
    const image: AmplienceImage = {
      altText: 'altText',
      image: {
        defaultHost: 'i1-qa.adis.ws',
        endpoint: 'bloblogltd',
        name: 'friends-shopping-bags',
        id: 'edc77f02-0e92-4b1a-88f9-711cacb5d650',
        mediaType: MediaType.IMAGE
      },
      src: '//i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags'
    };

    renderer.render(
      <Image altText={image.altText} src={image.src} dynamicImagingOptions={[{ w: 800 }, { w: 600 }, { w: 400 }]} />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('should render a lazy image with a height and width', (): void => {
    const image: AmplienceImage = {
      altText: 'altText',
      image: {
        defaultHost: 'i1-qa.adis.ws',
        endpoint: 'bloblogltd',
        name: 'friends-shopping-bags',
        id: 'edc77f02-0e92-4b1a-88f9-711cacb5d650',
        mediaType: MediaType.IMAGE
      },
      src: '//i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags'
    };

    renderer.render(<Image altText={image.altText} src={image.src} dynamicImagingOptions={[{ h: 800, w: 600 }]} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('should render a lazy image with a height, width and crop it', (): void => {
    const image: AmplienceImage = {
      altText: 'altText',
      image: {
        defaultHost: 'i1-qa.adis.ws',
        endpoint: 'bloblogltd',
        name: 'friends-shopping-bags',
        id: 'edc77f02-0e92-4b1a-88f9-711cacb5d650',
        mediaType: MediaType.IMAGE
      },
      src: '//i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags'
    };

    renderer.render(
      <Image altText={image.altText} src={image.src} dynamicImagingOptions={[{ h: 800, w: 600, sm: 'c' }]} />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
