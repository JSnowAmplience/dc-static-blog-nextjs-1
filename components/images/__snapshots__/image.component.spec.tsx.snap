// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render a lazy image with a height and width 1`] = `
<React.Fragment>
  <t
    afterLoad={[Function]}
    alt="altText"
    beforeLoad={[Function]}
    className="lazy-img"
    delayMethod="throttle"
    delayTime={300}
    effect=""
    placeholderSrc=""
    sizes=""
    src="https://i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags"
    srcSet="https://i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags?h=800&w=600 600w"
    threshold={100}
    visibleByDefault={false}
    wrapperClassName=""
  />
  <JSXStyle
    id="4022069848"
  >
    img.lazy-img{width:100%;}
  </JSXStyle>
</React.Fragment>
`;

exports[`Image should render a lazy image with a height, width and crop it 1`] = `
<React.Fragment>
  <t
    afterLoad={[Function]}
    alt="altText"
    beforeLoad={[Function]}
    className="lazy-img"
    delayMethod="throttle"
    delayTime={300}
    effect=""
    placeholderSrc=""
    sizes=""
    src="https://i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags"
    srcSet="https://i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags?h=800&w=600&sm=c 600w"
    threshold={100}
    visibleByDefault={false}
    wrapperClassName=""
  />
  <JSXStyle
    id="4022069848"
  >
    img.lazy-img{width:100%;}
  </JSXStyle>
</React.Fragment>
`;

exports[`Image should render a lazy image with width only 1`] = `
<React.Fragment>
  <t
    afterLoad={[Function]}
    alt="altText"
    beforeLoad={[Function]}
    className="lazy-img"
    delayMethod="throttle"
    delayTime={300}
    effect=""
    placeholderSrc=""
    sizes=""
    src="https://i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags"
    srcSet="https://i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags?w=800 800w,https://i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags?w=600 600w,https://i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags?w=400 400w"
    threshold={100}
    visibleByDefault={false}
    wrapperClassName=""
  />
  <JSXStyle
    id="4022069848"
  >
    img.lazy-img{width:100%;}
  </JSXStyle>
</React.Fragment>
`;

exports[`Image should render an image with a height, width and crop it to a defined poi 1`] = `
<React.Fragment>
  <t
    afterLoad={[Function]}
    alt="altText"
    beforeLoad={[Function]}
    className="lazy-img"
    delayMethod="throttle"
    delayTime={300}
    effect=""
    placeholderSrc=""
    sizes=""
    src="https://i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags"
    srcSet="https://i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags?h=800&w=600&sm=c&scaleFit=poi&poi=0.30741794586182,0.64965417250147,0.125,0.18867924528302 600w"
    threshold={100}
    visibleByDefault={false}
    wrapperClassName=""
  />
  <JSXStyle
    id="4022069848"
  >
    img.lazy-img{width:100%;}
  </JSXStyle>
</React.Fragment>
`;

exports[`Image should render an image with a height, width and crop it to the default poi 1`] = `
<React.Fragment>
  <t
    afterLoad={[Function]}
    alt="altText"
    beforeLoad={[Function]}
    className="lazy-img"
    delayMethod="throttle"
    delayTime={300}
    effect=""
    placeholderSrc=""
    sizes=""
    src="https://i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags"
    srcSet="https://i1-qa.adis.ws/i/bloblogltd/friends-shopping-bags?h=800&w=600&sm=c&scaleFit=poi&poi={$this.metadata.pointOfInterest.x},{$this.metadata.pointOfInterest.y},{$this.metadata.pointOfInterest.w},{$this.metadata.pointOfInterest.h} 600w"
    threshold={100}
    visibleByDefault={false}
    wrapperClassName=""
  />
  <JSXStyle
    id="4022069848"
  >
    img.lazy-img{width:100%;}
  </JSXStyle>
</React.Fragment>
`;
